prov/verbs/src/verbs_info.c

```diff
@@ -1192,6 +1261,145 @@ static int fi_ibv_get_matching_info(uint32_t version, const char *dev_name,
+				   src_addr->gid.global.subnet_prefix,
+				   src_addr->lid, src_addr->service);
+		} else {
+			dest_addr = addr;
+			VERBS_INFO(FI_LOG_CORE,
+				   "'%s' is taken as dest_addr: "
+				   "<gid = <interface_id=%x, "
+					   "subnet_prefix=%x>, "
+				   "lid = %d, service=%d>\n",
+				   node, dest_addr->gid.global.interface_id,
+				   dest_addr->gid.global.subnet_prefix,
+				   dest_addr->lid, dest_addr->service);
+		}
+		node = NULL;
+	}
+
+	if (!src_addr) {
+		src_addr = calloc(1, sizeof(*src_addr));
+		if (!src_addr) {
+			VERBS_INFO(FI_LOG_CORE,
+			           "failed to allocate src addr.\n");
+			return -FI_ENODATA;
+		}
+
+		/* TODO: handle */
+
+		if (flags & FI_SOURCE) {
+			if (service)
+				sscanf(service, "%" SCNu16, &src_addr->service);
+			VERBS_INFO(FI_LOG_CORE, "node '%s' service '%s' "
+				                "converted to <service=%d>\n",
+				   node, service, src_addr->service);
+		}
+	}
+
+	if (!dest_addr && node && !(flags & FI_SOURCE)) {
+		struct util_ns ns = (const struct util_ns){ 0 };
+		struct util_ns_attr ns_attr = {
+			.ns_port = fi_ibv_gl_data.dgram.name_server_port,
+			.name_len = sizeof(*dest_addr),
+			.service_len = sizeof(svc),
+			.service_cmp = fi_ibv_dgram_ns_service_cmp,
+			.is_service_wildcard = fi_ibv_dgram_ns_is_service_wildcard,
+		};
+
+		ret = ofi_ns_init(&ns_attr, &ns);
+		if (ret) {
+			VERBS_INFO(FI_LOG_CORE,
+				   "ofi_ns_init returns %d\n", ret);
+			ret = -FI_ENODATA;
+			goto out;
+		}
+
+		if (service)
+			svc = atoi(service);
+		svc0 = svc;
+		dest_addr = (struct ofi_ib_ud_ep_name *)
+			ofi_ns_resolve_name(&ns, node, &svc);
+		if (dest_addr) {
+			VERBS_INFO(FI_LOG_CORE,
+				   "'%s:%u' resolved to <gid <interface_id=%x, "
+				   "subnet_prefix=%x>, lid=%d>:%d\n",
+				   node, svc0, dest_addr->gid.global.interface_id,
+				   dest_addr->gid.global.subnet_prefix,
+				   dest_addr->lid, svc);
+		} else {
+			VERBS_INFO(FI_LOG_CORE,
+				   "failed to resolve '%s:%u'.\n", node, svc);
+			ret = -FI_ENODATA;
+			goto out;
+		}
+	}
+
+	for (iter_info = *info; iter_info; iter_info = iter_info->next) {
+		if (iter_info->ep_attr->type != FI_EP_DGRAM)
+			continue;
+		if (src_addr) {
+			iter_info->src_addr = calloc(1, sizeof(*src_addr));
+			memcpy(iter_info->src_addr, src_addr, sizeof(*src_addr));
+			iter_info->src_addrlen = sizeof(*src_addr);
+		}
+		if (dest_addr) {
+			iter_info->dest_addr = calloc(1, sizeof(*src_addr));
+			memcpy(iter_info->dest_addr, dest_addr, sizeof(*dest_addr));
+			iter_info->dest_addrlen = sizeof(*dest_addr);
+		}
+	}
+
+	if (src_addr)
+		free(src_addr);
+	if (dest_addr)
+		free(dest_addr);
+
+out:
+	return ret;
+}
+
+static int fi_ibv_handle_sock_addr(const char *node, const char *service,
+				   uint64_t flags, struct fi_info *hints,
+				   struct fi_info **info)
+{
+	struct rdma_cm_id *id = NULL;
+	struct rdma_addrinfo *rai;
+	int ret;
+
+	ret = fi_ibv_create_ep(node, service, flags, hints, &rai, &id);
```

Comment:
Why do we need to create ep again here? The rdma cm id can be passed to this function.

Reply:
Thanks, nice point. It's excessive